scripts/convert_tags_to_custom_properties/convert_tags_to_custom_properties.py

```diff
@@ -0,0 +1,206 @@
+import os
+import requests
+import json
+
+OPSLEVEL_API_TOKEN = os.environ["OPSLEVEL_API_TOKEN"]
+OPSLEVEL_ENDPOINT = "https://app.opslevel.com/graphql"
+
+LIST_CUSTOM_PROPERTIES_QUERY = """
+    query custom_service_properties($endCursor:String) {
+      account {
+        propertyDefinitions(after: $endCursor) {
+          pageInfo{
+            hasNextPage
+            endCursor
+          }
+          nodes {
+            name
+            aliases
+            schema
+            id
+          }
+        }
+      }
+    }
+"""
+
+SERVICES_BY_TAG_QUERY = """
+    query services_by_tag($endCursor: String, $tag_key:String){
+      account{
+        services(tag: {key: $tag_key}, after: $endCursor){
+          pageInfo{
+            hasNextPage
+            endCursor
+          }
+          nodes{
+            id
+            tags{
+              nodes{
+                id
+                key
+                value
+              }
+            }
+          }
+        }
+      }
+    }
+"""
+
+UPDATE_PROPERTY_MUTATION = """
+    mutation update_property($service_id:ID, $definition_alias:String, $value:JsonString!){
+      propertyAssign(input: {owner: {id: $service_id}, definition: {alias: $definition_alias},
+      value: $value, runValidation: false}) {
+        property{
+          value
+          owner{
+            ...on Service{
+              name
+            }
+          }
+        }
+        errors{
+          message
+          path
+        }
+      }
+    }
+"""
+
+
+def opslevel_graphql_query(query, variables=None):
+    headers = {
+        "Content-Type": "application/json",
+        "Authorization": f"Bearer {OPSLEVEL_API_TOKEN}",
+    }
+    data = {"query": query, "variables": variables}
+    response = requests.post(OPSLEVEL_ENDPOINT, json=data, headers=headers)
+    return response.json()
+
+
+def fetch_custom_properties():
+    """
+    Fetch custom properties from OpsLevel and return a list of properties.
+    """
+    cursor = None
+    has_next_page = True
+    properties = []  # Store fetched properties
+    while has_next_page:
+        response = opslevel_graphql_query(
+            LIST_CUSTOM_PROPERTIES_QUERY, variables={"endCursor": cursor}
+        )
+        nodes = response["data"]["account"]["propertyDefinitions"]["nodes"]
+        properties.extend(nodes)
+        page_info = response["data"]["account"]["propertyDefinitions"]["pageInfo"]
+        has_next_page = page_info["hasNextPage"]
+        cursor = page_info["endCursor"]
+
+    return properties
+
+
+def main():
+    properties = fetch_custom_properties()
+
+    for index, property_info in enumerate(properties, 1):
```

Comment:
Potential bug - What does `enumerate(properties, 1)` do here?

Reply:
This will print out the list of properties and have the index start at 1 for human readability (instead of choosing "index 0" for example)

Reply:
This will print out the list of properties and have the index start at 1 for human readability (instead of choosing "index 0" for example)

This:
```
% python convert_tags_to_custom_properties.py
1. Property Name: auto_deployed, Alias: auto_deployed, Schema Type: boolean
2. Property Name: Current Container Image, Alias: container_images, Schema Type: object
3. Property Name: data-class, Alias: data-class, Schema Type: array
4. Property Name: environment, Alias: environment, Schema Type: array
5. Property Name: Environments, Alias: environments, Schema Type: array
6. Property Name: Publicly Routable, Alias: publicly_routable, Schema Type: boolean
7. Property Name: VPCEnabled, Alias: vpc_enabled, Schema Type: boolean
```

Instead of this:
```
% python convert_tags_to_custom_properties.py
0. Property Name: auto_deployed, Alias: auto_deployed, Schema Type: boolean
1. Property Name: Current Container Image, Alias: container_images, Schema Type: object
2. Property Name: data-class, Alias: data-class, Schema Type: array
3. Property Name: environment, Alias: environment, Schema Type: array
4. Property Name: Environments, Alias: environments, Schema Type: array
5. Property Name: Publicly Routable, Alias: publicly_routable, Schema Type: boolean
6. Property Name: VPCEnabled, Alias: vpc_enabled, Schema Type: boolean
```